ui/app/AppLayouts/CommunitiesPortal/stores/CommunitiesStore.qml

```diff
@@ -85,6 +88,16 @@ QtObject {
         root.communitiesModuleInst.importCommunity(communityKey);
     }
 
+    function requestCommunityInfo(communityKey) {
```

Comment:
The name is `requestCommunityInfo` but what it really does is `fetchAndSpectate` ?

Reply:
No it actually **requests** community info and **if** there is any, then it'll get/fetch it and spectate.
There's a possibility of no success and no spectation.

Tried to stay consistent with the underlying API here

Reply:
I would expect `requestCommunityInfo` to only **request** the community info without the side effect of spectating it. In other words: `requestCommunityInfo.then(spectate(communityInfo.id))` looks as natural flow.

Reply:
So, what you're describing is actually what happens. First we request info, **then** when there's a signal with a response, whoever is interested in it will act on it. One part in particular with act on it by spectating.

We just don't have it wrapped in promise/future-like APIs